### ISSUE TITLE
Treat abstract methods as special cases in docparams

### ISSUE THREAD

**Issue by AWhetter** [PyCQA/pylint#2044]
### Steps to reproduce
With the docparams extension turned on, the following code raises the messages:

* "NotImplementedError" not documented as being raised [W9006]
* Redundant returns documentation [W9008]

```python
class BaseFilterClass(object):
    def check(self, item):
        """Check if the filter accepts the item.

        Args:
            item (str): The item to filter.

        Returns:
            bool: True if the filter accepts the item, False otherwise.
       """
       raises NotImplementedError
```

### Expected behavior
I think it's valid for this docstring to have returns documentation. You want to document what an implementation of the method should return. So W9008 should not be raised.

With W9006 I'm less certain. Ideally we would make sure that the docstring documents that the method is abstract, but that may happen in the text so we can't reliably detect it. I think my preference here is that W9006 would not get raised and we allow the user to choose whether to include it or not?

### pylint --version output

pylint 1.8.2, 
astroid 1.6.1
Python 2.7.3 (default, Jul 18 2012, 12:27:09) 
[GCC 4.4.6]
